provider/clusterrosaclassic/cluster_rosa_classic_resource.go

```diff
@@ -1021,6 +1040,20 @@ func (r *ClusterRosaClassicResource) Update(ctx context.Context, request resourc
 		return
 	}
 
+	clusterState := "Unknown"
+	if common.HasValue(state.State) && state.State.ValueString() != "" {
```

Comment:
Wondering why we need this part. Can't we just do 
if state.State.String() != string(cmv1.ClusterStateReady) or state.State.ValueString()  != string(cmv1.ClusterStateReady) ?

Reply:
If it's an empty string I prefer the error msg will be "unknown"